### PR TITLE
Java Lab: Encrypt validation field

### PR DESCRIPTION
**Pull request by molly-moen**
The validation field in Java Lab needs to be encrypted so students cannot see validation code. There were a couple changes required to make this work:
- change `validation` to `encrypted_validation` (encryption itself is handled by [serialized_properties](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/concerns/serialized_properties.rb#L66-L84)).
- add a new route to save start code. This was needed because the route currently in use updates properties by editing the hash directly, which does not allow us to take advantage of the encryption from `serialized_properties`. Now we will update `start_sources` and `validation` directly if both fields are present, otherwise we will set properties using the existing method.
- send the validation field when in `edit_blocks` mode

## Links
- spec: [Java Lab Validation](https://docs.google.com/document/d/14_ka0o1YFhAoXFl-WiyukrfIvT4C0IYIVVYcFaEmV7w/edit#)
- jira ticket: [JAVA-389](https://codedotorg.atlassian.net/browse/JAVA-389)

## Testing story
Tested locally and added unit tests. Ensured this worked for both Java lab and non-Java Lab levels, as app lab and game lab also use the levelbuilder save button component.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/levels_controller.rb [hunk 1]
```diff
@@ -270,6 +273,21 @@ def update
...
+  # Update start code for a level. If params contain validation and start_sources,
+  # set those directly to ensure encryption. Otherwise set via update_properties
+  def update_start_code
+    if !@level.respond_to?(:start_sources) || !@level.respond_to?(:validation)
+      return update_properties
+    end
+    @level.start_sources = params[:start_sources]
+    @level.validation = params[:validation]
+    @level.log_changes(current_user)
+    @level.save!
```
Comment (jmkulwik) [inline dashboard/app/controllers/levels_controller.rb]: Are there any other properties we'd want to update? i.e. perhaps we could always do the call to `update_properties` but update `start_sources` and `validation` and first then perhaps edit the params so they don't have those fields.

Comment (molly-moen) [reply dashboard/app/controllers/levels_controller.rb]: good idea, I'll try that out

Comment (molly-moen) [reply dashboard/app/controllers/levels_controller.rb]: updated!

**Review by jmkulwik — APPROVED** [review]
LGTM!

**Pull request merged**